cvat-ui/src/actions/plugins-actions.ts

```diff
@@ -39,17 +39,34 @@ export function checkPluginsAsync(): ThunkAction {
             DEXTR_SEGMENTATION: false,
         };
 
-        const promises: Promise<boolean>[] = [
-            PluginChecker.check(SupportedPlugins.ANALYTICS),
-            PluginChecker.check(SupportedPlugins.GIT_INTEGRATION),
-            PluginChecker.check(SupportedPlugins.DEXTR_SEGMENTATION),
-        ];
-
         try {
+            let pluginCheckError: any = null;
+
+            const promises: Promise<boolean>[] = [
+                PluginChecker.check(SupportedPlugins.ANALYTICS).catch((err) => {
+                    pluginCheckError = err;
+                    return false;
+                }),
+                PluginChecker.check(SupportedPlugins.GIT_INTEGRATION).catch(
+                    (err) => {
+                        pluginCheckError = err;
+                        return false;
+                    },
+                ),
```

Comment:
This cases are wrapped with `try catch` in plugin-checker so i don't think catches blocks are necessary here. Also you can dispatch `raisePluginCheckError` just here without having to re-throw an error.

Reply:
The catch statement does not dispatch `checkedAllPlugins` and even if it does it will not know what plugins are actually checked because the `Promise.all` will throw immediately after any promise returns a rejection error. This means that the screen just sits in the `fetching` state forever. My change allows each plugin to be checked before an alert is raised then it dispatches `checkedAllPlugins` with the finished plugin check list so that the page will load (set `fetching` to `false` and `initialized` to `true`) but still alert the user when one or more plugins fail.

I made the change because I am having issues getting nuclio to work but I haven't created an issue yet because I need to put some more time into it (and I think it is a nuclio issue). Seems to be my private ca certificate blocking nuclio from running properly.

Reply:
@TravisYeah, could you please report your problem with Nuclio? Even you cannot debug it on your side yet it is really important to identify all of them as early as possible.

Reply:
Instead of this construction I would suggest to use ``Promise.allSettled()`` instead of ``Promise.all()`` below that returns results of promises.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled